src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -698,12 +698,15 @@ private async Task<bool> PushTargets(IList<TargetSpecification> targets, TargetE
                     if (buildReason == TargetBuiltReason.BeforeTargets || buildReason == TargetBuiltReason.DependsOn || buildReason == TargetBuiltReason.None)
                     {
                         TargetEntry currentParent = parentTargetEntry;
+                        List<string> parentChain = new List<string>() { targetSpecification.TargetName };
```

Comment:
We were talking about possibly putting this in the `String.Equals` block below and figuring out the chain from scratch. In the normal (non-error) case, it would save a little bit of memory. I started an RPS run [here](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/284930/) to find out if that's a big deal or not.

Reply:
Looks like RPS passed so more likely to not be a big deal.

Reply:
It makes sense to me. Putting this to `String.Equals` block may consume more time to get the circular which should be fine.